File: clients/dart/cldf/lib/models/enums/media_source.dart
import 'package:json_annotation/json_annotation.dart';

/// Source of media
@JsonEnum(valueField: 'value')
enum MediaSource {
  /// File stored locally on device
  local('local'),

  /// File stored in cloud (iCloud, Google Photos, etc.)
  cloud('cloud'),

  /// Just a reference/URL, not actual file
  reference('reference'),

  /// Embedded in the archive itself
  embedded('embedded'),

  /// Legacy values for backward compatibility
  external('external'),
  photosLibrary('photos_library');

  const MediaSource(this.value);

  final String value;
}

File: clients/java/cldf-java/src/main/java/app/crushlog/cldf/models/enums/MediaSource.java
package app.crushlog.cldf.models.enums;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/** Source of media. */
public enum MediaSource {
  LOCAL("local"),
  CLOUD("cloud"),
  REFERENCE("reference"),
  EMBEDDED("embedded"),
  EXTERNAL("external"),
  // Legacy value for backward compatibility
  PHOTOS_LIBRARY("photos_library");

  private final String value;

  MediaSource(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @JsonCreator
  public static MediaSource fromValue(String value) {
    if (value == null) {
      return null;
    }
    for (MediaSource source : values()) {
      if (source.value.equalsIgnoreCase(value)) {
        return source;
      }
    }
    throw new IllegalArgumentException("Unknown media source: " + value);
  }
}
